torchgen/model.py

```diff
@@ -1240,8 +1242,8 @@ def __post_init__(self) -> None:
                 assert len(self.returns) == 0
 
         if self.arguments.tensor_options is not None:
-            assert self.kind() == SchemaKind.functional, (
-                "Found an operator that is not functional, but has tensor options arguments."
+            assert self.kind() == SchemaKind.functional or self.kind() == SchemaKind.out, (
+                "Found an operator that is not functional or out varuabt, but has tensor options arguments."
```

Comment:
This change seems suspicious - we don't have / want any out= ops to take in `TensorOptions` arguments.

We technically do have some factory functions that have out= variants ([example](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L628)), but you can see that they don't take in `TensorOptions` arguments - the metadata can be inferred from the `out=` tensor.

Reply:
bump :)